### PR TITLE
Fix issue where commands dont work if guild icon doesnt exist GH#3205

### LINKED ISSUE #3205: [BUG]: guild icon invalid
### Bot Version

4.0.1

### How are you hosting Modmail?

Other

### Error Logs

https://hastebin.cc/isepucoled

### Screenshots

_No response_

### Additional Information

The bot will not close threads if there is not an icon set for the server. 
It is failing to return the default icon.

### PR DESCRIPTION
**Pull request by Cordila**
Resolves #3205 
The bot will now use the bots display avatar if it fails to find a guild icon. 

Have tested this. 

_Ignore the commits_

### CODE REVIEW

File: bot.py [hunk 1]
```diff
@@ -913,7 +913,10 @@ async def process_dm_modmail(self, message: discord.Message) -> None:
                     color=self.error_color,
                     description=self.config["disabled_new_thread_response"],
                 )
-                embed.set_footer(text=self.config["disabled_new_thread_footer"], icon_url=self.guild.icon.url)
+                embed.set_footer(
+                    text=self.config["disabled_new_thread_footer"],
+                    icon_url=await self.bot.api.get_guild_icon(),
```
Comment (Jerrie-Aries) [inline bot.py:918]:
```suggestion
                    icon_url=await self.api.get_guild_icon(),
```

File: bot.py [hunk 2]
```diff
@@ -928,7 +931,7 @@ async def process_dm_modmail(self, message: discord.Message) -> None:
                 )
                 embed.set_footer(
                     text=self.config["disabled_current_thread_footer"],
-                    icon_url=self.guild.icon.url,
+                    icon_url=await self.bot.api.get_guild_icon(),
```
Comment (Jerrie-Aries) [inline bot.py:934]:
```suggestion
                    icon_url=await self.api.get_guild_icon(),
```

File: bot.py [hunk 3]
```diff
@@ -1335,7 +1338,7 @@ async def handle_react_to_contact(self, payload):
             )
             embed.set_footer(
                 text=self.config["disabled_new_thread_footer"],
-                icon_url=self.guild.icon.url,
+                icon_url=await self.bot.api.get_guild_icon(),
```
Comment (Jerrie-Aries) [inline bot.py:1341]:
```suggestion
                icon_url=await self.api.get_guild_icon(),
```

File: core/clients.py [hunk 4]
```diff
@@ -751,6 +751,11 @@ async def get_user_info(self) -> Optional[dict]:
                 }
             }
 
+    async def get_guild_icon(self):
+        if self.bot.guild.icon is None:
+            return self.bot.user.display_avatar.url
+        return self.bot.guild.icon.url
```
Comment (Jerrie-Aries) [inline core/clients.py:757]: To be honest, this should not be `async`.

But if this is changed, well, you have to remove all `await`s on the changes you made.

**Comment by Cordila** [comment]
Will remake PR on basis of suggestions by Jerrie as discussed in dev-chat

**Pull request closed without merging**